azurecontainerapps/azurecontainerapps.ts

```diff
@@ -177,30 +190,43 @@ export class azurecontainerapps {
                 console.log(tl.loc('CreateImageWithBuilderMessage'));
 
                 // Set the Oryx++ Builder as the default builder locally
-                new ContainerAppHelper().setDefaultBuilder();
+                appHelper.setDefaultBuilder();
 
                 // Create a runnable application image
-                new ContainerAppHelper().createRunnableAppImage(imageToDeploy, appSourcePath, runtimeStack);
+                appHelper.createRunnableAppImage(imageToDeploy, appSourcePath, runtimeStack);
+
+                // If telemetry is enabled, log that the builder scenario was targeted for this task
+                telemetryHelper.setBuilderScenario();
             }
 
             // If a Dockerfile was found or provided, create a runnable application image from that
             if (!util.isNullOrEmpty(dockerfilePath) && shouldBuildAndPushImage) {
                 console.log(tl.loc('CreateImageWithDockerfileMessage', dockerfilePath));
-                new ContainerAppHelper().createRunnableAppImageFromDockerfile(imageToDeploy, appSourcePath, dockerfilePath);
+                appHelper.createRunnableAppImageFromDockerfile(imageToDeploy, appSourcePath, dockerfilePath);
+
+                // If telemetry is enabled, log that the Dockerfile scenario was targeted for this task
+                telemetryHelper.setDockerfileScenario();
             }
 
             // Push image to Azure Container Registry
             if (shouldBuildAndPushImage) {
                 new ContainerRegistryHelper().pushImageToAcr(imageToDeploy);
+            } else {
+                // If telemetry is enabled, log that the previously built image scenario was targeted for this task
+                telemetryHelper.setImageScenario();
             }
 
             // Create or update Azure Container App
-            new ContainerAppHelper().createOrUpdateContainerApp(containerAppName, resourceGroup, imageToDeploy, optionalCmdArgs);
+            appHelper.createOrUpdateContainerApp(containerAppName, resourceGroup, imageToDeploy, optionalCmdArgs);
+            telemetryHelper.setSuccessfulResult();
```

Comment:
Should we have some Unsuccessful scenario as well? For example in the `catch` block below, maybe we could log an error?

Reply:
As mentioned in a previous comment, the default value of the `result` property tracked within `TelemetryHelper` is "failed", so unless it's explicitly marked as "successful" (like it is after `az containerapp up` is ran without errors), then we'd end up logging that the task failed somewhere in the flow.

Reply:
Actually, based on feedback given below, I've added a `setFailedResult()` method that is called in the `catch` block below as you suggested.